llvm/lib/Target/AMDGPU/SILoadStoreOptimizer.cpp

```diff
@@ -2055,10 +2055,16 @@ bool SILoadStoreOptimizer::promoteConstantOffsetToImm(
   if (!(MI.mayLoad() ^ MI.mayStore()))
```

Comment:
Just out of curiosity: why atomics are not supported?

Reply:
Atomics have identical addressing modes, they should be handled the same way